aiosmtpd/smtp.py

```diff
@@ -61,6 +65,7 @@ class _DataState(enum.Enum):
 EMPTYBYTES = b''
 MISSING = _Missing()
 NEWLINE = '\n'
+VALID_AUTHMECH = re.compile(r"[A-Z0-9_-]+\Z")
```

Comment:
We can revisit this as a bugfix if there is actually a SASL mechanism which has characters outside of these.